packages/react-native-codegen/src/parsers/parsers-primitives.js

```diff
@@ -193,16 +193,36 @@ function emitPromise(
   typeAnnotation: $FlowFixMe,
   parser: Parser,
   nullable: boolean,
+  types: TypeDeclarationMap,
+  aliasMap: {...NativeModuleAliasMap},
+  tryParse: ParserErrorCapturer,
+  cxxOnly: boolean,
+  translateTypeAnnotation: $FlowFixMe,
```

Comment:
I think we should keep the `emitPromise` signature as clean as possible.
Given that we are implementing it, I will ask to the caller of `emitPromise` to just pass the `elementType` we want to use as generic promise. This should simplify also the whole testing.

The `emitXXX` code should be as simple as possible. I look at them as the building blocks to emit the final type. Resolving the associated type is something that should happen before. What do you think?

Reply:
I don't think it is impossible, like `emitFunction`. It was clean because no one cares about the `T` inside, but now you have to keep processing it.